### DISCUSSION TITLE
Can `useHead` only be called on Server-Side？

### DISCUSSION THREAD

**Discussion by undermoonn** [nuxt/nuxt#16470] [General]
I try to generate some head informations on Server-Side with `useHead`. 
Because of some heavy denpencies, sending these large JS bundles to browser seems not to be a good operation.

There are my codes:
```js
<script setup>
if (process.server) {
  const options = {}
  // heavy denpencies and calculations
  useHead(options)
}
<script>
```

On Server-Side, it runs just fine. But after sending HTML to broswer, all meta tags was removed by runtime and replaced with default meta tags.

---

When initing nuxt `vueuse-head` plugin,  plugin will call `head.updateDOM` on `app:mounted` hook: https://github.com/nuxt/framework/blob/main/packages/nuxt/src/head/runtime/lib/vueuse-head.plugin.ts#L22

`head.updateDOM` will call `vueuse/head` `updateElements` function to remove existed meta tags:
https://github.com/vueuse/head/blob/main/src/dom/update-elements.ts#L70

---

Is it possible to disable this behavior on Client-Side? Or maybe I`m requesting a new feature 😀

**Comment by tylerforesthauser** [comment]
Would it be crazy to maybe use a useState ref for this? Since useHead can accept a ref as its argument, just create a useState ref, assign your calculated head object to it only on the server and then sit back and watch as useState carries your head data safely to the client side 😜

**Comment by undermoonn** [comment]
```js
<script setup>
if (process.server) {
  const { heavyCalculation } = await import('~/path/to/util')
  const state = useState('my-meta', () => heavyCalculation(/** data from api request */))
  useHead({ ...state.value })
} else {
  const state = useState('my-meta')
  useHead({ ...state.value })
}
<script>
```

I try to use `useState`, it works just like I expected!

In this case, `heavyCalculation` will not bundled in client JS. Thanks a lot! @tylerforesthauser
